src/data/fintype.lean

```diff
@@ -480,38 +480,27 @@ instance Prop.fintype : fintype Prop :=
 def set_fintype {α} [fintype α] (s : set α) [decidable_pred s] : fintype s :=
 fintype.subtype (univ.filter (∈ s)) (by simp)
 
-
-/-! ### pi -/
-
-/-- A dependent product of fintypes, indexed by a fintype, is a fintype. -/
-instance pi.fintype {α : Type*} {β : α → Type*}
-  [decidable_eq α] [fintype α] [∀a, fintype (β a)] : fintype (Πa, β a) :=
-@fintype.of_equiv _ _
-  ⟨univ.pi $ λa:α, @univ (β a) _,
-    λ f, finset.mem_pi.2 $ λ a ha, mem_univ _⟩
-  ⟨λ f a, f a (mem_univ _), λ f a _, f a, λ f, rfl, λ f, rfl⟩
-
 namespace fintype
 
-variables [fintype α] [decidable_eq α] {δ : α → Type*} [decidable_eq (Π a, δ a)]
+variables [fintype α] [decidable_eq α] {δ : α → Type*}
```

Comment:
Sometimes these "unneeded" instances in arguments help Lean to apply a lemma in cases when `decidable_eq (Π a, δ a)` is not the one you're expecting.

Reply:
The instance isn't actually used at all, I got rid of the use of `finset.image`, and used `finset.map`, so I don't think that it needs to synthesize an instance anywhere.

Reply:
Thanks for the explanation.

Reply:
Here the instance does not appear in the statement of the lemma when you replace `image` with `map` as Chris did, so removing it looks like a good move (and if you add it back then the linter complains it is unneeded).